apps/web/utils/getAllTemplates.ts

```diff
@@ -1,5 +1,12 @@
-import templateData from './template-data.json';
+import templatesData from "./template-data.json";
+
+export interface TemplateInterface {
```

Comment:
Using "interface" postfix is against TS convention. It can simply be Template

For array types you don't need to create an external type. You can simply use Template[] as type whenever you deal with arrays.